lnwallet/musig_session_test.go

```diff
@@ -0,0 +1,247 @@
+package lnwallet
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr/musig2"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/stretchr/testify/require"
+)
+
+// nodeType is an enum that represents the two nodes in our test harness.
+type nodeType uint8
+
+const (
+	// nodeAlice is the node that initiates the session.
+	nodeAlice nodeType = iota
+
+	// nodeBob is the node that responds to the session.
+	nodeBob
+)
+
+type muSessionHarness struct {
+	aliceCommit *wire.MsgTx
+	bobCommit   *wire.MsgTx
```

Comment:
I think we should make small changes to the commits on each round just to ensure that the correct commits are being signed etc

Reply:
Yep done.